specifics/Lambda/Lambda.cpp

```diff
@@ -466,119 +458,220 @@ void Lambda::read_distortionCorrection(Tango::Attribute &attr)
+ *	Read attribute energyThreshold related method
  *	Description: set/get first energy threshold in eV.<br>
  *               The photon is counted If the energy is above this threshold.<br>
  *
- *	Data type:	Tango::DevFloat
+ *	Data type:	Tango::DevDouble
  *	Attr type:	Scalar
  */
 //--------------------------------------------------------
-void Lambda::read_energyThreshold1(Tango::Attribute &attr)
+void Lambda::read_energyThreshold(Tango::Attribute &attr)
 {
-	DEBUG_STREAM << "Lambda::read_energyThreshold1(Tango::Attribute &attr) entering... " << endl;
-	/*----- PROTECTED REGION ID(Lambda::read_energyThreshold1) ENABLED START -----*/
-    try
+	DEBUG_STREAM << "Lambda::read_energyThreshold(Tango::Attribute &attr) entering... " << endl;
+	/*----- PROTECTED REGION ID(Lambda::read_energyThreshold) ENABLED START -----*/
+	try
     {
-        *attr_energyThreshold1_read = m_camera->getThresholdEnergy(0);
-        attr.set_quality(Tango::ATTR_VALID);
-    	attr.set_value(attr_energyThreshold1_read);
+		double w_energyThreshold =0.0;
+		m_camera->getEnergyThreshold(w_energyThreshold);
+		*attr_energyThreshold_read = w_energyThreshold;
+		attr.set_value(attr_energyThreshold_read);
     }
     catch (Tango::DevFailed& df)
     {
-        manage_devfailed_exception(df, "read_energyThreshold1");
+        manage_devfailed_exception(df, "read_energyThreshold");
     }
-	
-	/*----- PROTECTED REGION END -----*/	//	Lambda::read_energyThreshold1
+	catch(...)
+    {
+        INFO_STREAM      << "Read_energyThreshold FAILED" << endl;
+	}
+
+	/*----- PROTECTED REGION END -----*/	//	Lambda::read_energyThreshold
 }
 //--------------------------------------------------------
 /**
- *	Write attribute energyThreshold1 related method
+ *	Write attribute energyThreshold related method
  *	Description: set/get first energy threshold in eV.<br>
  *               The photon is counted If the energy is above this threshold.<br>
  *
- *	Data type:	Tango::DevFloat
+ *	Data type:	Tango::DevDouble
  *	Attr type:	Scalar
  */
 //--------------------------------------------------------
-void Lambda::write_energyThreshold1(Tango::WAttribute &attr)
+void Lambda::write_energyThreshold(Tango::WAttribute &attr)
 {
-	DEBUG_STREAM << "Lambda::write_energyThreshold1(Tango::WAttribute &attr) entering... " << endl;
+	DEBUG_STREAM << "Lambda::write_energyThreshold(Tango::WAttribute &attr) entering... " << endl;
 	//	Retrieve write value
-	Tango::DevFloat	w_val;
+	Tango::DevDouble	w_val;
 	attr.get_write_value(w_val);
-	/*----- PROTECTED REGION ID(Lambda::write_energyThreshold1) ENABLED START -----*/
-    try
+	/*----- PROTECTED REGION ID(Lambda::write_energyThreshold) ENABLED START -----*/
+	try
     {
-        float data = static_cast<float>(w_val);
-        m_camera->setThresholdEnergy(0, data);
-        yat4tango::PropertyHelper::set_property(this, "MemorizedEnergyThreshold1", w_val);
+		double data = static_cast<double>(w_val);
+        m_camera->setEnergyThreshold(data);
```

Comment:
à remplacer par:
m_camera->setEnergyThreshold(w_val);

et:
il faut memoriser la valeur via:
yat4tango::PropertyHelper::set_property(this, "MemorizedEnergyThreshold", w_val);